### PR TITLE
matrix initialization organization

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
Organizes matrix initialization, taking command into RegisterTile and IServerSpawn initializations after the matrix is ready.
Fixes movable matrixes having objects in the wrong position for clients.
Sets electric manager to start running when the round starts.

EDIT:
Fixes mapped objects breaking if their transform.position.z was tampered with.
Fixes the chance of matrix IDs being desynced.
Fixes servers initializing their custom transform twice.
The matrix manager will no longer wait for load with a coroutine to set itself as initialized, but use the ScenesLoaded event, from here it will start the atmos systems and trigger the IServerSpawn from all matrices.
Radiation and atmos threads will now start after roundstart.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/NetworkedMatrix.cs [hunk 1]
```diff
@@ -32,6 +32,10 @@ public class NetworkedMatrix : MonoBehaviour
 		[HideInInspector]
 		public MatrixSync MatrixSync;
 
+		[HideInInspector] public bool Initialized;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/NetworkedMatrix.cs]: this looks more like a ``NonSerialized``

**Comment by Aranclanos** [comment]
Having some issues with piping and firelocks dnm

**Comment by Aranclanos** [comment]
atmos systems problems solved! but I just noticed a client doesnt update positions properly with quantum pads, so I need a lil bit more time

**Comment by Aranclanos** [comment]
ready for player testing

**Comment by corp-0** [comment]
first play test (after opening unity twice) 
image

```
NullReferenceException: Object reference not set to an instance of an object
RegisterTile.OnStartServer () (at Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterTile.cs:175)
Mirror.NetworkIdentity.OnStartServer () (at Assets/Mirror/Runtime/NetworkIdentity.cs:660)
UnityEngine.Debug:LogException(Exception, Object)
Mirror.NetworkIdentity:OnStartServer() (at Assets/Mirror/Runtime/NetworkIdentity.cs:664)
Mirror.NetworkServer:SpawnObject(GameObject, NetworkConnection) (at Assets/Mirror/Runtime/NetworkServer.cs:984)
Mirror.NetworkServer:Spawn(GameObject, NetworkConnection) (at Assets/Mirror/Runtime/NetworkServer.cs:996)
Mirror.NetworkServer:SpawnObjects() (at Assets/Mirror/Runtime/NetworkServer.cs:1081)
<LoadSubScene>d__37:MoveNext() (at Assets/Scripts/Managers/SubSceneManager/SubSceneManager.cs:64)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr, Boolean&)
```

This is before the countdown is done, editor play test.

**Comment by corp-0** [comment]
looks like it happens when loading Ancient Station

**Comment by corp-0** [comment]
image
these objects in lavaland aren't children of lavaland matrix

**Comment by corp-0** [comment]
I'm not entirely sure but I feel like sounds that should be muffled, like the ones behind walls, aren't muffled

**Comment by Aranclanos** [comment]
those errors are already fixed in #7380, just update your local

**Comment by Aranclanos** [comment]
more stuff appeared, dont merge!

**Comment by ThatDan123** [comment]
> more stuff appeared, dont merge!

Dont worry we cant speed merge anymore :p

**Comment by ThatDan123** [comment]
Has this fixed #7392 ? (assuming its not a mapping issue)

**Comment by corp-0** [comment]
> those errors are already fixed in #7380, just update your local

I pulled and tested this pr

**Comment by Aranclanos** [comment]
declaring this ready for the third time

**Comment by Aranclanos** [comment]
edited OP with new changes that I added in new commits

**Comment by ThatDan123** [comment]
> declaring this ready for the third time

Just one codacy issue

**Review by ThatDan123 — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**